lib/hal_client/representation.rb

```diff
@@ -294,6 +315,40 @@ def raw
 
     MISSING = Object.new
 
+    def flatten_section(section_hash)
+      result = Array.new
+
+      section_hash.each_pair do |rel, rel_data|
+        if rel_data.is_a?(Array)
+          rel_data.each do |single_item|
+            result << { rel: rel, data: single_item }
+          end
+        elsif rel_data.is_a?(Hash)
+          result << { rel: rel, data: rel_data };
+        end
+      end
+
+      result
+    end
+
+    def link_from_link_entry(hash_entry)
+      rel = hash_entry[:rel]
+      hash_data = hash_entry[:data]
+      href = hash_data['href']
+
+      if hash_data['templated']
+        Link.new(rel: rel, template: Addressable::Template.new(href))
+      else
+        Link.new(rel: rel, target: Representation.new(hal_client: hal_client, href: href))
+      end
```

Comment:
it might be better to do this in the `Link` class. Ie, allow it to take a raw(ish) json link hash and a hal client object and let it figure this stuff out.